src/sagas/pushNotification.js

```diff
@@ -445,6 +469,77 @@ export function* dismissOptInQuestion() {
   yield STORE.setItem(pushNotificationKey.optInDismissed, true);
 }
 
+/**
+ * This function retrieves the tx details from the wallet history.
+ * @param {Object} wallet the current wallet
+ * @param {string} txId the tx id
+ * @returns {Promise<{
+ *  tx: { txId: string, timestamp: number, voided: boolean },
+ *  tokens: { uid: string, name: string, symbol: string, balance: number, isRegistered: boolean }[]
+ * }>} the tx details
+ * @example
+ * {
+ *   tx: {
+ *     txId: '000021e7addbb94a8e43d7f1237d556d47efc4d34800c5923ed3a75bf5a2886e',
+ *     timestamp: 1673039453,
+ *     voided: false,
+ *   },
+ *   tokens: [
+ *     {
+ *       uid: '00',
+ *       name: 'Hathor',
+ *       symbol: 'HTR',
+ *       balance: 500,
+ *       isRegistered: true,
+ *     }
+ *   ],
+ */
+export const getTxDetails = async (wallet, txId) => {
+  const txTokensBalance = await wallet.getTxById(txId);
+  const [tx] = txTokensBalance;
+  const txDetails = {
+    tx: {
+      txId: tx.txId,
+      timestamp: tx.timestamp,
+      voided: tx.voided,
+    },
+    tokens: txTokensBalance.map((each) => ({
+      uid: each.tokenId,
+      name: each.tokenName,
+      symbol: each.tokenSymbol,
+      balance: each.balance,
+      isRegistered: !!tokens.tokenExists(each.tokenId),
+    })),
+  };
+  return txDetails;
+};
+
+export function setInitialNotificationData(notification) {
```

Comment:
Why do you need this method? A docstring would be useful here.

Reply:
When the application is in the background or quit state it can not use the internal storage of the application because it was not loaded yet. Therefore, I hold the notification in the storage, and when the application initializes it checks this data. If it is present, the details modal is triggered.